src/vs/workbench/contrib/tasks/browser/abstractTaskService.ts

```diff
@@ -576,7 +577,10 @@ export abstract class AbstractTaskService extends Disposable implements ITaskSer
 		await this._extensionService.whenInstalledExtensionsRegistered();
 
 		await Promise.all(
-			this._getActivationEvents(type).map(activationEvent => this._extensionService.activateByEvent(activationEvent))
+			this._getActivationEvents(type).map(activationEvent => Promise.race([this._extensionService.activateByEvent(activationEvent), timeout(2000).then(() => {
```

Comment:
The suggestion in https://github.com/microsoft/vscode/pull/176298 is a bit simpler, a single race for the promise of all activation events, vs a race per activation event.